Content.Client/StationEvents/StationEventManager.cs

```diff
@@ -0,0 +1,40 @@
+﻿#nullable enable
+using Content.Shared.StationEvents;
+using Robust.Shared.Interfaces.Network;
+using Robust.Shared.IoC;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Content.Client.StationEvents
+{
+    class StationEventManager : SharedStationEvent, IStationEventManager
+    {
+#pragma warning disable 649
+        [Dependency] private readonly IClientNetManager _netManager = default!;
+#pragma warning restore 649
+        private List<string>? _events;
+        public List<string>? StationEvents
+        {
+            get
+            {
+                if (_events == null)
+                    RequestEvents();
+                return _events;
+            }
+            set => _events = value;
+        }
```

Comment:
I think there's Lazy<T> that does this as well.

Reply:
I don't really know if Lazy<T> is usable here. The problem is that I don't wait for the response, but return null and then set the value later (which is not possible with Lazy, cause Lazy.Value is not settable).